xc/xenops_server_xen.ml

```diff
@@ -1468,6 +1467,32 @@ module VM = struct
 			)
 		| FD fd -> f fd
 
+	let wait_ballooning task vm =
+		on_domain
+			(fun xc xs _ _ di ->
+				let domid = di.Xenctrl.domid in
+				let balloon_active_path = xs.Xs.getdomainpath domid ^ "/control/balloon-active" in
+				let balloon_active =
+					try
+						Some (xs.Xs.read balloon_active_path)
+					with _ -> None
+				in
+				match balloon_active with
+				(* Not Windows or Windows is not currently ballooning *)
```

Comment:
We should abstract from "Windows" here: a balloon driver that advertises this xenstore key can be in any kind of guest.